include/igraph_constants.h

```diff
@@ -60,7 +60,7 @@ typedef enum { IGRAPH_RECIPROCITY_DEFAULT = 0,
              } igraph_reciprocity_t;
 
 typedef enum { IGRAPH_ADJ_DIRECTED = 0,
-               IGRAPH_ADJ_UNDIRECTED = 1, IGRAPH_ADJ_MAX = 1,
```

Comment:
Renumbering only breaks ABI, not API, so it should have no drawbacks at all on `develop`.

Reply:
Not quite; if the caller happened to hardcode the enum value in the calling code (which shouldn't have happened, but most compilers only treat this as a warning), then this could break. This is not as far-fetched as it seems; take a look at `types-RR.yaml` in `igraph/rigraph`, which does this all the time (unfortunately). Luckily enough the code for `igraph_weighted_adjacency()` is not used there, but let's not get into the habit of randomly reassigning numeric values of enums just because it formally does not break the API.

Reply:
Yes, but that's the caller's problem, and it _should_ really be fixed there ... In the Mathematica interface I also pass integers from Mma to C code, which represent these choices. But there is always an explicit `switch` which maps these integers (which are a concern of the Mma interface only) to proper named enum values. For example: https://github.com/szhorvat/IGraphM/blob/master/IGraphM/LibraryResources/Source/IG.h#L57

I'm not saying that we should just go ahead and renumber anything without a good justification, but this should really really be fixed on the R/Stimulus side. In the end, Stimulus needs to be made aware of the enum value names that we use in C. There's no other way that's reliable. If it operates with integers at the moment that's not okay.